### PR TITLE
Improved readme

### PR DESCRIPTION
**Pull request by Raffo**
@szuecs Please provide comments, especially on the change of logic that you are working on.

### CODE REVIEW

**Comment by hjacobs** [comment]
We also need to describe how to deploy Skipper.

**Comment by hjacobs** [comment]
Maybe it's easier to put the Kubernetes manifests as YAML files into a folder? I would expect the README to just contain something like:

```
kubectl apply -f deploy-folder-whatever/ingress-controller-deployment.yaml
kubectl apply -f deploy-folder-whatever/skipper-daemonset.yaml

**Comment by Raffo** [comment]
@hjacobs Added additional information. Sounds good, I will move the files to files.

File: README.md [hunk 1]
```diff
@@ -5,14 +5,14 @@
...
-It should be deployed once per Kubernetes cluster. It will use the EC2 instance metadata to find the CloudFormation
-stack it belongs to. This value is then used to discover the required AWS resources that are attached to each newly 
-created Application Load Balancer.
+It will use the EC2 instance metadata to find the CloudFormation
+stack it belongs to which is then used to discover the required AWS resources that are attached to each newly created Application Load Balancer.
 
 ## How it works
 
-It continuously polls the API server checking for ingress resources. For each cycle it creates load balancers for new
+The controller continuously polls the API server checking for ingress resources. For each cycle it creates load balancers for new
```
Comment (szuecs) [inline README.md]: This part is about how the kubernetes part works, should we also write how the aws part works?
We find load balancers by certificate and have ownership by tag. This is maybe something the user does not expect.

File: README.md [hunk 2]
```diff
@@ -57,5 +82,239 @@ Due to the 32 characters limitation, it can be truncated.
 When the controller detects that a managed load balancer for the current cluster doesn't have a matching ingress 
 resource anymore, it deletes all the previously created resources.
 
+## Building
+
+This project provides a `Makefile` that can be used to build a binary or to build a docker image for it. To build a binary for your current operating system, it is enough to run `make` or `make build.linux` to build an image for Linux. 
+To create a docker image, you can execute `make build.docker` instead. 
+
+## Deploy
+
```
Comment (szuecs) [inline README.md]: Maybe explain what will be deployed. You deploy 3 components: the ingress controller as deployment, the ingress loadbalancer skipper as daemonset  and mate as external DNS provider, which will point ingress spec hostnames to the selected ALB name (IIRC got from ingress status) as ALIAS record.

Comment (Raffo) [inline README.md]: I will not describe mate, just use put it as a nice to have. The other two are described already.

File: README.md [hunk 3]
```diff
@@ -57,5 +82,239 @@ Due to the 32 characters limitation, it can be truncated.
...
+      - {CidrIp: 0.0.0.0/0, FromPort: 80, IpProtocol: tcp, ToPort: 80}
+      - {CidrIp: 0.0.0.0/0, FromPort: 443, IpProtocol: tcp, ToPort: 443}
+      VpcId: "$VPC_ID"
+      Tags:
+        - Key: "ClusterID"
+          Value: "$CLUSTER_ID"
+    Type: AWS::EC2::SecurityGroup
+```
+
+- Make the port used by `skipper` (in this example, port `9999`) accessible from the IP range of the VPC to allow health checking from the Application Load Balancer.
```
Comment (szuecs) [inline README.md]: Maybe also tell that skipper provides metrics for all ingress backends via http(":9911/metrics/servehost")

Comment (Raffo) [inline README.md]: I would prefer to keep this minimal. This is the README for the ingress controller and I would leave to other documents (skipper doc itself?) this description.

File: README.md (hunk 1 above)
Comment (Raffo) [inline README.md]: It's a good idea, please contribute it to this readme.

File: README.md (hunk 2 above)
Comment (lmineiro) [inline README.md]: Describing skipper as part of the overall solution and not mate seems incomplete. We do mention hostnames.

**Comment by hjacobs** [comment]
:+1:

File: README.md [hunk 4]
```diff
@@ -57,5 +82,54 @@ Due to the 32 characters limitation, it can be truncated.
 When the controller detects that a managed load balancer for the current cluster doesn't have a matching ingress 
 resource anymore, it deletes all the previously created resources.
 
+## Building
+
+This project provides a `Makefile` that can be used to build a binary or to build a docker image for it. To build a binary for your current operating system, it is enough to run `make` or `make build.linux` to build an image for Linux. 
+To create a docker image, you can execute `make build.docker` instead.
```
Comment (mikkeloscar) [inline README.md]: > run `make` or `make build.linux` to build an image for Linux

nit: it's a binary, not an image :)

Comment (Raffo) [inline README.md]: True, copy paste.

**Comment by szuecs** [comment]
:+1:

**Comment by Raffo** [comment]
👍

**Pull request merged**